### ISSUE TITLE
mupdf: bad alpha values

### ISSUE THREAD

**Issue by chengcjun** [dothinking/pdf2docx#96]
pdf2docx convert 2020.pdf 2020.dcox
[2020.pdf](https://github.com/dothinking/pdf2docx/files/6628731/2020.pdf)

mupdf: bad alpha values

output nothing

**Comment by dothinking** [comment]
抱歉这么长时间都没有处理你这个问题。

目前这个库能力有限，难以处理你的PDF那样杂志级别的复杂排版和丰富样式。推荐一个在线的转换网站，目前看来应该是技术最好的：

https://smallpdf.com/pdf-to-word

当然，我会把转换杂志级别的PDF作为`pdf2docx`的长期目标。

**Comment by yangxiaomin08** [comment]
any update on this?

**Comment by dothinking** [comment]
> any update on this?

Rework on this issue again and found it did be a bug. Some images in pdf file are composed of a base image and a mask. Same size is required when merge them together, while in this case, there exists one image with different size from its mask.

Now, it's fixed by checking the size before merging mask. For now, the bad image will be ignored if the size doesn't match.

The fix will be released soon with new version `v0.5.3`.

**Comment by dothinking** [comment]
The code to find the image causing the issue:

```python
>>> import fitz
>>> doc = fitz.open('2020.pdf')
>>> page = doc[67]  # the page with abnormal image
>>> for item in page.get_images():
...     xref, mask_xref = item[:2]
...     if mask_xref==0: continue   # image without mask
...     pix = fitz.Pixmap(doc, xref)
...     mask = fitz.Pixmap(doc, mask_xref)
...     if pix.width!=mask.width and pix.height!=mask.height:
...         print(item)
...

(1436, 1708, 11, 12, 8, 'DeviceRGB', '', 'X819', 'JPXDecode')
```
